src/commands/generate.ts

```diff
@@ -0,0 +1,161 @@
+    console.log(input);
+    const [paramName, paramValue] = input.split(/=(.+)/, 2);
+    console.log(paramName, paramValue);
+    this._params[String(paramName)] = paramValue;
+  }
+
+  private disableHooksParser = (input: string) => {
+    const [hookType, hookNames] = input.split(/=/);
+    if (!hookType) { throw new Error('Invalid --disable-hook flag. It must be in the format of: --disable-hook <hookType> or --disable-hook <hookType>=<hookName1>,<hookName2>,...'); }
+    if (hookNames) {
+      this._disableHooks[String(hookType)] = hookNames.split(',');
+    } else {
+      this._disableHooks[String(hookType)] = true;
+    }
+  }
+
+  private mapBaseURLParser = (input: string) => {
+    const re = /(.*):(.*)/g; // NOSONAR
+    let mapping: any[] | null = [];
+    if ((mapping = re.exec(input)) === null || mapping.length !== 3) {
+      throw new Error('Invalid --map-base-url flag. A mapping <url>:<folder> with delimiter : expected.');
+    }
+
+    this._mapBaseUrlToFolder.url = mapping[1].replace(/\/$/, '');
+    this._mapBaseUrlToFolder.folder = path.resolve(mapping[2]);
+
+    const isURL = /^https?:/;
+    if (!isURL.test(this._mapBaseUrlToFolder.url.toLowerCase())) {
+      throw new Error('Invalid --map-base-url flag. The mapping <url>:<folder> requires a valid http/https url and valid folder with delimiter `:`.');
+    }
+  }
+}
+
+export default class Generate extends Command {
+  static description = 'Generator is a tool that you can use to generate whatever you want basing on the AsyncAPI specification file as an input.';
+
+  static examples: Example[] = [
+    'asyncapi generate asyncapi.yaml @asyncapi/html-template'
+  ];
+
+  static flags = {
+    help: Flags.help({ char: 'h' }),
+    'disable-hook': Flags.string({
+      char: 'd',
+      description: 'disable a specific hook type or hooks from a given hook type',
+      multiple: true
+    }),
+    install: Flags.boolean({
+      char: 'i',
+      default: false,
+      description: 'installs the template and its dependencies (defaults to false)'
+    }),
+    debug: Flags.boolean({
+      description: 'to enable specific errors in the console'
+    }),
+    'no-overwrite': Flags.string({
+      char: 'n',
+      multiple: true,
+      description: 'glob or path of the file(s) to skip when regenerating'
+    }),
+    output: Flags.string({
+      char: 'o',
+      description: 'directory where to put the generated files (defaults to current directory)'
+    }),
+    'force-write': Flags.boolean({
+      default: false,
+      description: 'force writing of the generated files to given directory even if it is a git repo with unstaged files or not empty dir (defaults to false)'
+    }),
+    'watch-tempalte': Flags.boolean({
+      description: 'watches the template directory and the AsyncAPI document, and re-generate the files when changes occur. Ignores the output directory.'
+    }),
```

Comment:
we should use `watch` flag from it https://github.com/asyncapi/cli/blob/master/src/flags.ts#L3

Reply:
We have a custom description in the watch flag for the generator command, I think we can convert [`watchFlag`](https://github.com/asyncapi/cli/blob/8fe89e2d122517775905d80531de3a0edc8f6244/src/flags.ts#L3) into a function that would return the watch flag then we can have custom description.

Reply:
If you want, you can convert `watchFlag` to the function, but if you don't please only change name of flag to the `--watch` and add fallback `-w`